CRM/Extension/Manager/Payment.php

```diff
@@ -86,8 +86,8 @@ public function onPreInstall(CRM_Extension_Info $info) {
         throw new CRM_Core_Exception(ts('Billing mode in info file has wrong value.'));
     }
 
-    $dao->is_recur = trim($info->typeInfo['isRecur']);
-    $dao->payment_type = trim($info->typeInfo['paymentType']);
+    $dao->is_recur = trim($info->typeInfo['isRecur'] ?? '');
+    $dao->payment_type = trim($info->typeInfo['paymentType'] ?? '');
```

Comment:
hmm - if `$dao->save();` can't handle NULL we are gonna have a lot of problems - might need to fix  `$dao->save();

Reply:
No the problem is not with $dao->save() the problem is the `trim()` passing in NULL values to trim() triggers deprecation notices in php8.1

Reply:
ah - of course